### PR TITLE
Adds the banner component from Calypso.

### PR DESCRIPTION
**Pull request by zinigor**
This PR adapts the banner component:
* The Banner component is now available.
* The plan icons are now available.
* The plan constants and some helper functions are now available and can be used to get rid of hardcoded plan slugs.

What has been left out:
* The plan price component has not been ported yet because I didn't know if it was needed.
* The dismissable card component that has been also ported isn't actually dismissable on its own, it needs to be removed from the parent component.

### CODE REVIEW

File: client/components/dismissible-card/style.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.dismissible-card__close-icon {
```
Comment (MichaelArestad) [inline client/components/dismissible-card/style.scss]: Needs the `dops-` prefix.

File: client/components/plans/plan-icon/style.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.plan-icon {
```
Comment (MichaelArestad) [inline client/components/plans/plan-icon/style.scss]: Needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+	.dops-banner__prices {
+		display: flex;
+		justify-content: flex-start;
+
+		.plan-price {
+			margin-bottom: 0;
+		}
+
+		.plan-price.is-discounted,
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: Needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		.plan-price.is-discounted .plan-price__currency-symbol {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: Both needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+			color: $gray-dark;
+		}
+
+		.has-call-to-action & .plan-price {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: `.plan-price` needs `dops-` prefix

File: client/components/banner/style.scss [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+	font-size: 12px;
+	margin: 8px 0 0 0;
+	text-align: left;
+	width: 100%;
+
+	.dops-banner__prices {
+		display: flex;
+		justify-content: flex-start;
+
+		.plan-price {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: Needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+			display: block;
+		}
+	}
+}
+
+.dops-banner__icon-plan {
+	display: flex;
+	margin-right: 16px;
+
+	.plan-icon {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: Needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+		color: $blue-wordpress;
+		display: flex;
+	}
+
+	&:hover {
+		transition: all 100ms ease-in-out;
+		&.is-card-link {
+			box-shadow: 0 0 0 1px $gray, 0 2px 4px lighten( $gray, 20% );
+		}
+		.card__link-indicator {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: I think this needs `dops-` prefix.

File: client/components/banner/style.scss [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+		@include banner-color( $alert-yellow );
+	}
+	&.is-upgrade-premium {
+		@include banner-color( $alert-green );
+	}
+	&.is-upgrade-business {
+		@include banner-color( $alert-purple );
+	}
+
+	.card__link-indicator {
```
Comment (MichaelArestad) [inline client/components/banner/style.scss]: I think this needs `dops-` prefix.

**Comment by zinigor** [comment]
Added prefixes, thanks for taking a look!

File: client/lib/plans/constants.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { compact, includes } from 'lodash';
```
Comment (eliorivero) [inline client/lib/plans/constants.js]: Please update this import to be
```
import compact from 'lodash/compact';
import includes from 'lodash/includes';
```

**Comment by eliorivero** [comment]
Apart from that comment about the `import` LGTM 🐑

**Pull request merged**